packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -651,5 +662,5 @@ function getPossibleAmiTypes(instanceTypes: InstanceType[]): NodegroupAmiType[]
     throw new Error('instanceTypes of different architectures is not allowed');
   }
 
-  return archAmiMap.get(Array.from(architectures)[0])!;
+  return archAmiMap.get(Array.from(architectures)[0])!.concat(NodegroupAmiType.CUSTOM);
```

Comment:
Instead of concat here, should we just add it directly to the x8664AmiTypes, arm64AmiTypes, etc. Otherwise this line of code would be easily missed and maintainers and other contributors may not realize those amiTypes are not accurate and complete.